.github/workflows/tox.yml

```diff
@@ -54,11 +54,16 @@ on:
         description: Command to run after test commands.
         required: false
         type: string
+      publish_pypi:
+        default: false
+        description: Whether to publish to PyPI
+        required: false
+        type: boolean
 # keep permissions at top level because this is a composite workflow
 permissions:
   checks: read
   contents: read
-  id-token: write
+  id-token: write # release
```

Comment:
Setting the OIDC privilege is highly discouraged because giving such elevated privileges to random scripts like transitive build dependencies, testing automation and other things can lead to privilege elevation and impersonation attacks and who knows what else. For this reason, I emphasize this in all the docs for `pypi-publish` that are in my reach. We even managed to update GitHub's own PyPI+OIDC doc to show that it's only acceptable in the publishing job that only downloads the dists and calls `pypi-publish` but nothing else.
So whenever I see `id-token: write` anywhere outside of the narrow scope of a single job doing this single thing, I know immediately that it's a security vulnerability.